components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -923,7 +923,30 @@ def render_image_region(request, iid, z, t, conn=None, **kwargs):
                     if tile_length > max_tile_length:
                         tile_size[i] = max_tile_length
                 w, h = tile_size
-            level = levels-int(zxyt[0])
+            v = int(zxyt[0])
+            if v < 0:
+                logger.debug(
+                    "render_image_region: invalid level %s" % v, exc_info=True
+                )
+                return HttpResponseBadRequest('Invalid resolution')
+
+            if levels == 0:  # non pyramid file
+                if v > 0:
+                    logger.debug(
+                        "render_image_region: invalid level %s" % v,
+                        exc_info=True
+                    )
+                    return HttpResponseBadRequest('Invalid resolution')
+                else:
+                    level = None
+            else:
+                level = levels-v
+                if level < 0:
+                    logger.debug(
+                        "render_image_region: invalid level %s" % v,
+                        exc_info=True
+                    )
+                    return HttpResponseBadRequest('Invalid resolution')
```

Comment:
Probably useful to include the resolution and reason in the response, `'Invalid resolution level, %d > number of available levels %d' % (v, levels)`

Depending on what you want to do you might use the same message in both the log statement and the response. There is also no reason to include context (`render_image_region`) in the message as that is included, along with the line number, in the standard log format.